src/pudl_archiver/archivers/phmsagas.py

```diff
@@ -66,18 +66,18 @@ async def get_zip_resource(
         For example: annual_underground_natural_gas_storage_2017_present.zip
         """
         url = f"https://www.phmsa.dot.gov/{link}"
-        file = str(match.group(1)).replace("-", "_")  # Get file name
+        filename = str(match.group(1)).replace("-", "_")  # Get file name
 
         # Set dataset partition
-        dataset = "_".join(file.lower().split("_")[0:-2])
+        dataset = "_".join(filename.lower().split("_")[0:-2])
 
         if dataset not in PHMSA_DATASETS:
             logger.warning(f"New dataset type found: {dataset}.")
 
         # Set start year
-        start_year = int(file.split("_")[-2])
+        start_year = int(filename.split("_")[-2])
 
-        download_path = self.download_directory / f"phmsagas_{file}.zip"
+        download_path = self.download_directory / f"{self.name}-{filename}.zip"
```

Comment:
All of the other archivers use a file naming pattern like:

`dataset-part1-part2.ext`

where:
- dataset is the name of the dataset (e.g. ferc1, eia923, mshamines)
- part1 is the value of the first partition dimension contained in
  the file (e.g. ca, ny, wy, for states, 2019 if it's a year)
- part2 is the value of the second partition dimension contained in
  the file, if applicable.
- ext is the file name extension indicating file type.

I've changed the naming for phmsagas to follow the `dataset-*` part of the naming convention, but have left the filename in place rather than using only the dataset and start year, since if we just use the start year it looks like years are missing in the archive, as each file contains several years of data.

The `dataset` and `start_year` partitions are still used in datapackage.json to refer to the individual resources for programmatic purposes.

Reply:
I agree with the decision not to use start date for the filenames here. But in this case the expected output here is a file like: `phmsagas-annual_underground_natural_gas_storage_2017_present.zip` ?

Would it be worth reversing the substitution from "-" to "_" in line 69 to result in filenames like ``phmsagas-annual-underground-natural-gas-storage-2017-present.zip``? Having both "-" and "_" is a bit visually confusing here and at least this would bring us towards having a consistent snake case filename, even if the structure is slightly different than those of other archivers.

Reply:
I wasn't totally thrilled about this combo either. It also occurs in the MSHA archives (though there we do use the actual partition values, rather than the filename).  Really we don't want anybody parsing these filenames directly, and they should be using the metadata in `datapackage.json` to address individual partitions. I guess even what we have right now isn't literally the original filenames since we're changing the one with dashes to underscores. We haven't had to deal with this data-table based partitioning before. Previously it was either one file per timeslice (with individual tables inside it) or in the case of EPA CEMS, files that were all part of the same table, but partitioned the data based on values (year and state) within that one table.

I just wanted to somehow differentiate the `phmsagas` part of the name from the other parts, since it's a different kind of thing, and it was possible to follow the same convention as all the FERC and EIA data.

Another thing that we're seeing come up is overlap in partitions where what differentiates them is the file format, e.g. years where both XBRL and DBF data are available in FERC, which seems like another type of partitioning we might want to work with and make obvious in the filenames and partition metadata.